result/bgpv4prefix.yaml

```diff
@@ -0,0 +1,73 @@
+components:
+  schemas:
+    Bgpv4Prefix.State.Request:
+      description: >-
+        The request to retrieve IPv4 Prefix state of a BGPv4 Peer.
+      type: object
+      properties:
+        bgp_peer_names:
+          description: >-
+            The names of BGPv4 Peers for which IPv4 Prefix state information
+            will be retrieved. If no names are specified then the results will
+            contain IPv4 Prefix state for all configured BGPv4 Peers.
+          type: array
+          items:
+            type: string
+          x-constraint:
+            - "/components/schemas/Bgp.v4Peer/properties/name"
+
+    Bgpv4Prefix.State:
+      description: >-
+        BGP peer IPv4 Prefix state.
+      type: object
+      properties:
+        bgp_peer_name:
+          description: >-
+            The name of the BGPv4 Peer associated with the IPv4 prefix state
+            information.
+          type: string
+        prefixes:
+          type: array
+          items:
+            $ref: '#/components/schemas/Bgpv4Prefix.State.Prefix'
+
+    Bgpv4Prefix.State.Prefix:
+      description: >-
+        BGP IPv4 prefix information
+      type: object
+      properties:
+        ipv4_address:
+          description: >-
+            The IPv4 prefix.
+          type: string
+          format: ipv4
+        prefix_length:
+          description: >-
+            The length of the IPv4 prefix.
+          type: integer
+        ipv4_next_hop:
+          description: >-
+            The IPv4 address of the egress interface.
+          type: string
+          format: ipv4
+        ipv6_next_hop:
+          description: >-
+            The IPv6 address of the egress interface.
+          type: string
+          format: ipv6
+        origin:
+          description: >-
+            The origin of the prefix.
+          type: string
+          enum: [igp, egp]
+        path_id:
```

Comment:
path_id and even as_path might be optional. Is it expected that optional attribs will just not be present in the response ? Should this be added to documentation of such attributes. ( most will be optional except prefix/next-hop)

Reply:
openapi attributes are by default all optional unless marked as required in the object